packages/caching-compiler/caching-compiler.js

```diff
@@ -281,12 +281,12 @@ CachingCompiler = class CachingCompiler extends CachingCompilerBase {
     const cacheMisses = [];
     const arches = this._cacheDebugEnabled && Object.create(null);
 
-    return Promise.all(inputFiles.map(async (inputFile) => {
+    inputFiles.forEach(inputFile => {
```

Comment:
I'm honestly kind of angry how much of a performance difference switching from `Promise.all(async (inputFile) => ...)` to `inputFiles.forEach(inputFile => ...)` makes, here.